tests/Avalonia.Markup.Xaml.UnitTests/Xaml/EventTests.cs

```diff
@@ -48,6 +65,43 @@ public void Exception_Is_Thrown_If_Event_Not_Found()
             XamlTestHelpers.AssertThrowsXamlException(() => AvaloniaRuntimeXamlLoader.Load(xaml, rootInstance: target));
         }
 
+
+
+        [Fact]
+        public void Attached_Event_Routed_Event_Handler()
+        {
+            var xaml = @"<Panel xmlns='https://github.com/avaloniaui' Button.Click='OnClick'><Button Name='target'/></Panel>";
+            var host = new MyPanel();
+
+            AvaloniaRuntimeXamlLoader.Load(xaml, rootInstance: host);
+
+            var target = host.FindControl<Button>("target");
+            target.RaiseEvent(new RoutedEventArgs
+            {
+                RoutedEvent = Button.ClickEvent,
+            });
+
+            Assert.True(host.WasClicked);
+        }
+
+        [Fact]
+        public void Attached_Event_Tunnel_Routed_Event_Handler()
+        {
+            var xaml = @"<Panel xmlns='https://github.com/avaloniaui' PreviewKeyDown='OnKeyDown'><TextBox Name='target'/></Panel>";
```

Comment:
I don't think this PR should include "Preview" tunnel handling as a new feature.

Reply:
See https://github.com/AvaloniaUI/Avalonia/discussions/13628